controllers/services_controller.go

```diff
@@ -0,0 +1,144 @@
+	return &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      MetricsServiceName,
+			Namespace: namespace,
+			Labels: map[string]string{
+				"prometheus.ssp.kubevirt.io": "true",
+			},
+			Finalizers: []string{finalizerName},
+		},
+		Spec: v1.ServiceSpec{
+			InternalTrafficPolicy: &policyCluster,
+			IPFamilies:            []v1.IPFamily{v1.IPv4Protocol},
+			IPFamilyPolicy:        &familyPolicy,
+			Ports: []v1.ServicePort{
+				{
+					Name:       metrics.MetricsPortName,
+					Port:       443,
+					Protocol:   v1.ProtocolTCP,
+					TargetPort: intstr.FromString(metrics.MetricsPortName),
+				},
+			},
+			Selector: map[string]string{
+				"prometheus.ssp.kubevirt.io": "true",
+				"name":                       "ssp-operator",
+			},
+			SessionAffinity: v1.ServiceAffinityNone,
+			Type:            v1.ServiceTypeClusterIP,
+		},
+	}
+}
+
+// Annotation to generate RBAC roles to watch and update Pods
+// +kubebuilder:rbac:groups="",resources=services,verbs=get;watch;list;create;update;delete
+
+func CreateServiceController(mgr controllerruntime.Manager) *serviceReconciler {
+	client := mgr.GetClient()
+
+	reconciler := NewServiceReconciler(client)
+
+	return reconciler
+}
+
+func StartServiceController(mgr controllerruntime.Manager, reconciler *serviceReconciler) error {
+	err := reconciler.createMetricsService()
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+
+	return reconciler.setupController(mgr)
+}
+
+func (r *serviceReconciler) createMetricsService() error {
+	namespace, err := GetOperatorNamespace(r.log)
+	if err != nil {
+		return err
+	}
+
+	service := ServiceObject(namespace)
+
+	return r.client.Create(context.TODO(), service)
+}
+
+func (r *serviceReconciler) setupController(mgr ctrl.Manager) error {
+	predicate, _ := predicate.LabelSelectorPredicate(metav1.LabelSelector{MatchLabels: map[string]string{"prometheus.ssp.kubevirt.io": "true"}})
+	return ctrl.NewControllerManagedBy(mgr).
+		Named("service-controller").
+		For(&v1.Service{}).
+		WithEventFilter(predicate).
+		Complete(r)
+}
+
+// serviceReconciler reconciles the required services in the operator's namespace
+type serviceReconciler struct {
+	client           client.Client
+	log              logr.Logger
+	subresourceCache common.VersionCache
+}
+
+func NewServiceReconciler(client client.Client) *serviceReconciler {
+	return &serviceReconciler{
+		client:           client,
+		log:              ctrl.Log.WithName("controllers").WithName("Resources"),
+		subresourceCache: common.VersionCache{},
+	}
+}
+
+var GetOperatorNamespace = func(logger logr.Logger) (string, error) {
+	nsBytes, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+	if err != nil {
+		return "", err
+	}
+	ns := strings.TrimSpace(string(nsBytes))
+	logger.Info("Found namespace", "Namespace", ns)
+	return ns, nil
+}
+
+func (r *serviceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.Result, err error) {
```

Comment:
You can use `controllerutil.CreateOrUpdate()` to simplify this function. It was used for the SSP operator, but now it was copied and extended: 
https://github.com/kubevirt/ssp-operator/blob/1b07ce5934fc78a14d3bfc3522cd01d2ab7f563d/internal/common/resource.go#L285-L286